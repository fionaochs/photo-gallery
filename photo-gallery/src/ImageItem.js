import React, { Component } from 'react'

export default class Creature extends Component {
    render () {
        return <li>

            <div id="info-container">
                <h2>{this.props.creature.title}</h2>
                <p className="creature-type">{this.props.creature.description}</p>
            </div>

            <div className="image-container">
                <img 
                    alt={this.props.creature.url}
                    src={this.props.creature.url} />
            </div>

            <p className="horns">{this.props.creature.horns}</p>
        </li>
    }
}
//render 
//for every creature selected
//<li>
//   <div id=info> <h2/> <p/> </div>
//   <div class img> <img/> </div>
// </li>
//will be inserted into <ul>