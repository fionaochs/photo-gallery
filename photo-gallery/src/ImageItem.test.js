import ImageItem from './ImageItem.js';
import React from "react";
import renderer from "react-test-renderer";
import images from './data.js';

test('renders ImageItem.js correctly', () => {
    const tree = renderer
      .create(<ImageItem creature={images[0]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });