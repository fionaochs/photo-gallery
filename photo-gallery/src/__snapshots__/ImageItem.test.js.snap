// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageItem.js correctly 1`] = `
<li>
  <div
    id="info-container"
  >
    <h2>
      UniWhal
    </h2>
    <p
      className="creature-type"
    >
      A unicorn and a narwhal nuzzling their horns
    </p>
  </div>
  <div
    className="image-container"
  >
    <img
      alt="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
      src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
    />
  </div>
  <p
    className="horns"
  >
    1
  </p>
</li>
`;
